### PR TITLE
Add create user endpoint

### LINKED ISSUE #70: Add create user endpoint

### LINKED ISSUE #73: A user has an employer they belong_to
Re: https://github.com/WorkHands/RAPiDSkills/issues/72#issuecomment-549503163

Let's just do this formally. Let's include a `employer_id` so we can attach a user to an organization as part of the create user endpoint https://github.com/WorkHands/RAPiDSkills/issues/70.

We should likely support accepting a name or id.

### PR DESCRIPTION
**Pull request by littleforest**
closes #70 
closes #73 

## Description
- Add Employer association to user (organization)
- Add create user endpoint

## QA
Using Postman:
- [x] Create user through API endpoint ([see docs](https://rapid-skills-staging-pr-74.herokuapp.com/v1/docs/#create-user))
- [x] Create user without email, should get error

### CODE REVIEW

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class API::V1::UsersController < API::V1::APIController
+  def create
+    @user = User.where(
+      email: params[:data][:attributes][:email]
+    ).first_or_initialize(password: SecureRandom.uuid) # tmp pw for now
```
Comment (Pcushing) [inline app/controllers/api/v1/users_controller.rb]: Can we formalize this so we skip a password for these users? Or do something more explicit to make it clear there's a user without their own generated password here? 

I'm worried that we can't easily identify users from users without account passwords they created right now (unless we already solved this and i'm just not remembering).

File: app/serializers/api/v1/user_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class API::V1::UserSerializer
+  include FastJsonapi::ObjectSerializer
+  attributes :email,
+             :name,
+             :organization_title
```
Comment (Pcushing) [inline app/serializers/api/v1/user_serializer.rb]: Can we include their `employer` and serialize the `employer` with `title` instead? 

Mainly, i'm curious how this is handled in this gem for comparison to my experience with active model serializers.

File: app/controllers/api/v1/users_controller.rb (hunk 1 above)
Comment (littleforest) [reply app/controllers/api/v1/users_controller.rb]: How about I create a user role called "lead"? And then we can filter on those if needed.

File: spec/requests/api/v1/users_spec.rb [hunk 3]
```diff
@@ -0,0 +1,176 @@
+require 'rails_helper'
+
+RSpec.describe API::V1::UsersController, type: :request do
+  describe "POST #create" do
+    let(:path) { "/v1/users" }
+
+    before { allow(SecureRandom).to receive(:uuid).and_return("password") }
+
+    context "with valid params" do
```
Comment (Pcushing) [inline spec/requests/api/v1/users_spec.rb]: It'd be great to make a shared rspec resource for some of the standard things we'll always be checking such as statuses, creates with valid params, etc.

File: app/serializers/api/v1/user_serializer.rb (hunk 2 above)
Comment (littleforest) [reply app/serializers/api/v1/user_serializer.rb]: Sure -- should I have the FE send `employer_title` instead? You had `organization_name` in #72 , but I was worried that these different field names for request/response were going to cause confusion.

File: spec/requests/api/v1/users_spec.rb (hunk 3 above)
Comment (Pcushing) [reply spec/requests/api/v1/users_spec.rb]: This isnt necessary here, but let's use the next endpoint as an excuse to do so.

File: app/controllers/api/v1/users_controller.rb (hunk 1 above)
Comment (Pcushing) [reply app/controllers/api/v1/users_controller.rb]: 👍 great. As long as there's something we can work with so we know they dont really have a password later when they try to sign up.

File: app/serializers/api/v1/user_serializer.rb (hunk 2 above)
Comment (Pcushing) [reply app/serializers/api/v1/user_serializer.rb]: This is what i was thinking:

```
{
 email: 'email here',
 name: 'name here',
 employer: {
  title: 'title here',
  other_attrs: 'here'
 }
}
```

**Comment by littleforest** [comment]
@Pcushing ready for re-review. The json-api spec doesn't allow for the format you suggested with the employer; instead there are `relationships` and `included` sections [per the spec](https://jsonapi.org/format/#document-compound-documents).  @jhash I *think* I'm returning the data correctly by specifying the Employer record type as an Organization, but let me know if anything seems weird on the front-end side once you start implementing this.

**Comment by Pcushing** [comment]
@littleforest works for me. I'll have to dig in a bit to better understand the duplication in `relationships` and `included` but that's fine.

**Pull request merged**